File: Codigos_por_competidor/Mariana_Pineda/CountingSort.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
  /*
	 * Autor: mariana 
	 * Problema : Counting Sort
	 * Juez online: omegaUp
	 * Veredicto: 66.67%
	 * URL:https://omegaup.com/arena/problem/Counting-Sort/
	 */
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] numeritos = new int[n];
		for (int i = 0; i < numeritos.length; i++) {
			numeritos[i] = sc.nextInt();
		}

		Arrays.sort(numeritos);

		for (int i = 0; i < numeritos.length; i++) {
			System.out.print(numeritos[i] + " ");
		}
		System.out.println();
    int aux = 0;
		for (int i = 0; i < numeritos.length; i++) {
      aux = 0;
			for (int j = 0; j < i; j++) {
				if (numeritos[i] > numeritos[j]) {
					aux++;
				}
			}
			System.out.print(aux + " ");
		}
		System.out.println();
	}
}

File: Resueltos_por_competidor/Tomás_Triana_Galvis/CountingSort.cpp
#include <bits/stdc++.h>
typedef long long ll;
using namespace std;

/*
 * Autor: Tomás Triana Galvis
 * Problema: Counting Sort
 * Juez online: OmegaUp 11398
 * Veredicto: Accepted
 * Url: https://omegaup.com/arena/problem/Counting-Sort/#problems
 **/ 

int main()
{
    int n,a;
    cin>>n;
    vector<int> arr(n), ans(n);
    for(int i = 0; i<n;i++){
        cin>>arr[i];
    }
    sort(arr.begin(),arr.end());
    int j;
    for(int i=0;i<arr.size();i++){
        j = i;
        ans[i] = i;
        while(arr[i]==arr[i+1]){
            ans[i+1] = j;
            i++;
        }

    }

    for(auto an:arr){
        cout<<an<<" ";
    }
    cout<<endl;
    for(auto an:ans){
        cout<<an<<" ";
    }
    return 0;
}
